agent/cache/cache.go

```diff
@@ -376,6 +376,13 @@ func (c *Cache) getEntryLocked(
 	// Check if re-validate is requested. If so the first time round the
 	// loop is not a hit but subsequent ones should be treated normally.
 	if !tEntry.Opts.Refresh && info.MustRevalidate {
+		if entry.Fetching {
```

Comment:
Ah, this is one of those lines that I hope doesn't surface code that was relying on side effects here.

AFAIK, (and I don't know much)  there's not a lot of cache types that actually use `MustRevalidate`.

So my gut tells me that this should work beautifully for the bug in https://github.com/hashicorp/consul/issues/12048 🤞🏼 

But it would feel good to check in with @banks or @freddygv if they have more context here to get a third/ forth pair of eyes 💯